lib/domain/model/session.dart

```diff
@@ -154,6 +160,10 @@ class Credentials {
   @HiveField(2)
   final UserId userId;
 
+  /// Object of the currently authenticated [MyUser].
+  @HiveField(3)
+  final MyUser? user;
```

Comment:
1) First of all, if you're adding `MyUser`, then why do you leave `UserId` there? `MyUser` already contains its `UserId`, thus you might just do a getter for it.

2) However, I don't think `MyUser` should be added to the `Credentials` model, as it is useless there from the business logic point. `Credentials` are about tokens and `UserId` of who those tokens are. `MyUser` there will always be outdated. Also you're adding that field there just to be able to move the `MyUser` fetched to the `MyUserRepository` basically. Only that breaks the domain layer logic: you're adding a model to domain, which isn't a domain requirement, thus shouldn't be there in domain at all.